src/components/facets/FacetRow.tsx

```diff
@@ -8,9 +9,15 @@ export interface IFacetRowProps extends React.ClassAttributes<FacetRow> {
   isChecked: boolean;
 }
 
+export const MAX_NAME_LENGTH: number = 25;
+
 export class FacetRow extends React.Component<IFacetRowProps, any> {
 
   render() {
+    const label: JSX.Element = this.props.facetRow.formattedName.length > MAX_NAME_LENGTH
+      ? <Tooltip title={this.props.facetRow.formattedName} placement='top'>{this.props.facetRow.formattedName}</Tooltip>
+      : <span>{this.props.facetRow.formattedName}</span>;
```

Comment:
I don't know if it's pertinent, but adding a '...' a the end when the string is cut could be nice for the user.